config/checkstyle-input-suppressions.xml

```diff
@@ -54,7 +54,6 @@
             files="[\\/]checks[\\/]imports[\\/]unusedimports[\\/]InputUnusedImportsShadowed.java"/>
   <suppress id="noCheckstyleInInputs"
             files="[\\/]checks[\\/]imports[\\/]redundantimport[\\/]InputRedundantImportWithChecker.java"/>
-  <!-- until https://github.com/checkstyle/checkstyle/issues/4845 -->
   <suppress id="noCheckstyleInInputs"
           files="[\\/]packageobjectfactory[\\/]bar[\\/]FooCheck.java"/>
```

Comment:
Issue is valid until under comment there is `id="noCheckstyleInInputs"`
Or you try to say that they need to be in permanent suppression, if yes please explain.

Reply:
yes, these are considered under permanent suppression, **PackageObjectFactoryTest** is dependent on FooCheck.java which is extending an AbstractCheck.java, and this import is needed **import com.puppycrawl.tools.checkstyle.api.AbstractCheck;** in all 4 suppressions

Reply:
Confirmed https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/packageobjectfactory/abc/FooCheck.java